packages/shared-components/src/components/core/onto-dropdown/onto-dropdown.tsx

```diff
@@ -0,0 +1,123 @@
+import {Component, Event, h, State, Prop, EventEmitter, Listen} from '@stencil/core';
+import {DropdownOption} from '../../../models/dropdown-option';
+import {TranslationService} from '../../../services/translation.service';
+
+/**
+ * A reusable dropdown component built using StencilJS. This component supports configurable labels, tooltips, icons,
+ * and items, making it versatile for various use cases. It also integrates with a translation service to handle
+ * internationalization.
+ */
+@Component({
+    tag: 'onto-dropdown',
+    styleUrl: 'onto-dropdown.scss',
+    shadow: false,
+})
+export class OntoDropdown {
+
+    /**
+     * Indicates whether the dropdown menu is open.
+     */
+    @State() open = false;
+
+    /**
+     * The label displayed on the dropdown button.
+     * Overrides `nameLabelKey` if provided.
+     */
+    @Prop() nameLabel: string;
+
+    /**
+     * Key for the dropdown button label, used with the translation service.
+     */
+    @Prop() nameLabelKey: string;
+
+    /**
+     * The tooltip text displayed for the dropdown button.
+     * Overrides `tooltipLabelKey` if provided.
+     */
+    @Prop() tooltipLabel: string;
+
+    /**
+     * Key for the tooltip label, used with the translation service.
+     */
+    @Prop() tooltipLabelKey: string;
+
+    /**
+     * Icon class for the main dropdown button.
+     */
+    @Prop() iconClass = '';
+
+    /**
+     * Array of dropdown options. Each option includes a value, label, tooltip, and optional icon class.
+     */
+    @Prop() items: DropdownOption[];
+
+    /**
+     * Event emitted when a dropdown item is selected.
+     * The event payload contains the value of the selected item.
+     */
+    @Event() valueChanged: EventEmitter<string>;
```

Comment:
And here the payload is string, although the value option is set to be `any`. You need to clear this up